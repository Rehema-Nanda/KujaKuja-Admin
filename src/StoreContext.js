// https://mobx-react.js.org/recipes-context

import React from "react";
import PropTypes from "prop-types";
import RootStore from "./stores/RootStore";

export const StoreContext = React.createContext(null);

export const StoreProvider = ({ children }) => {
    const store = new RootStore();
    return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};
StoreProvider.propTypes = {
    children: PropTypes.node.isRequired,
};

export const StoreConsumer = StoreContext.Consumer;

export const useStore = () => {
    const store = React.useContext(StoreContext);
    if (!store) {
        // this is especially useful in TypeScript so you don't need to be checking for null all the time
        throw new Error("useStore must be used within a StoreProvider.");
    }
    return store;
};
