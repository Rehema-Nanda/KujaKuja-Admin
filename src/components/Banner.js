import React from "react";
import PropTypes from "prop-types";
import {
    Container, Row, Col, Button,
} from "reactstrap";

import { observer } from "mobx-react";

import "../index.css";
import { StoreContext } from "../StoreContext";

class Banner extends React.PureComponent {
    render() {
        const { userStore: { userInfo } } = this.context;
        const { pageTitle, elementAction, buttonFunction, buttonDisabled } = this.props;

        const title = pageTitle.charAt(0).toUpperCase() + pageTitle.slice(1);
        let buttonLabel;

        if (elementAction === "list") {
            buttonLabel = `Add ${title}`;
        }
        else if (elementAction === "update") {
            buttonLabel = `Delete ${title}`;
        }

        return (

            <Container>
                <Row>
                    <Col sm={{ size: 7 }} md={{ size: 8 }}>
                        <div className="page-title">
                            <h1 className="left">{title}</h1>
                            <p>
                                Manage your
                                {" "}
                                {pageTitle}
                            </p>
                        </div>
                    </Col>
                    <Col sm={{ size: 5 }} md={{ size: 4 }}>
                        <div className="page-title">
                            {
                                userInfo
                                && userInfo.length > 0
                                && userInfo[0].isAdmin
                                && elementAction !== "add"
                                && buttonLabel
                                && (
                                    <Button disabled={buttonDisabled} color="primary" className="primary" onClick={buttonFunction}>
                                        {buttonLabel}
                                    </Button>
                                )
                            }
                        </div>
                    </Col>
                </Row>
            </Container>

        );
    }
}

Banner.propTypes = {
    pageTitle: PropTypes.string.isRequired,
    elementAction: PropTypes.string.isRequired,
    buttonFunction: PropTypes.func.isRequired,
    buttonDisabled: PropTypes.bool,
};

Banner.defaultProps = {
    buttonDisabled: false,
};

Banner.contextType = StoreContext;

export default observer(Banner);
