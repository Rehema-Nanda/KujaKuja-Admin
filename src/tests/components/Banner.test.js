import React from "react";
import { mount, shallow } from "enzyme";
import { StoreProvider } from "../../StoreContext";
import Banner from "../../components/Banner";

describe("Banner component", () => {
    const testProps = {
        elementAction: "list",
        pageTitle: "Countries",
        buttonFunction: jest.fn(),
    };
    const wrapper = mount(<Banner {...testProps} />, { wrappingComponent: StoreProvider }); // eslint-disable-line react/jsx-props-no-spreading

    it("renders correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    // TODO: get these tests to work again
    it.skip("should display delete text on button", () => {
        const newProps = {
            ...testProps,
            elementAction: "update",
        };
        const tree = shallow(<Banner {...newProps} />); // eslint-disable-line react/jsx-props-no-spreading
        expect(tree.find("Button").text()).toBe("Delete Country");
    });

    it.skip("should execute function on button click", () => {
        wrapper.find("Button").simulate("click");
        expect(testProps.buttonFunction).toHaveBeenCalled();
    });
});
