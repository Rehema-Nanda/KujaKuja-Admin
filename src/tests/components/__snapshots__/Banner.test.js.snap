// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component renders correctly 1`] = `
"<Banner elementAction=\\"list\\" pageTitle=\\"Countries\\" buttonFunction={[Function: mockConstructor]} buttonDisabled={false}>
  <Container tag=\\"div\\">
    <div className=\\"container\\">
      <Row tag=\\"div\\">
        <div className=\\"row\\">
          <Col sm={{...}} md={{...}} tag=\\"div\\" widths={{...}}>
            <div className=\\"col-sm-7 col-md-8\\">
              <div className=\\"page-title\\">
                <h1 className=\\"left\\">
                  Countries
                </h1>
                <p>
                  Manage your
                   
                  Countries
                </p>
              </div>
            </div>
          </Col>
          <Col sm={{...}} md={{...}} tag=\\"div\\" widths={{...}}>
            <div className=\\"col-sm-5 col-md-4\\">
              <div className=\\"page-title\\" />
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </Container>
</Banner>"
`;
